overlord/ifacestate/handlers.go

```diff
@@ -96,9 +96,15 @@ func (m *InterfaceManager) doSetupProfiles(task *state.Task, _ *tomb.Tomb) error
 		return err
 	}
 	if len(affectedSnaps) == 0 {
-		affectedSnaps = append(affectedSnaps, snapInfo)
+		affectedSnaps = append(affectedSnaps, snapName)
 	}
-	for _, snapInfo := range affectedSnaps {
+	for _, snapName := range affectedSnaps {
+
+		snapInfo, err := snapstate.Current(task.State(), snapName)
+		if err != nil {
+			return err
+		}
+		snap.AddImplicitSlots(snapInfo)
```

Comment:
Or, for a more realistic approach, I'd like to keep this as-is and have a separate branch that ensures we just call AddImplicitSlots in one place in snapstate.